src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
```

Comment:
If we're just going to `unwrap` the option (i.e. assume it's non-empty), the argument should not be an `Option`, but just a `&'static mut [u8]`. This forces the client to deal with it if it's _actually_ optional, and removes ambiguity otherwise.